README.md

```diff
@@ -1,9 +1,71 @@
 # terraform-scripts
 
 This repository holds our common Terraform Open edX deployment scripts.
+They can be used to provision and manage the infrastructure required by an Open edX instance on AWS.
 
-Currently it actually contains two different ways to provision AWS resources
+### TODOS:
+- [ ] Write usage documentation
+- [ ] Expand the conventions section below with more details and examples
+
+## Architecture map
+
+See [this document](https://docs.google.com/drawings/d/1s2hd6hJSKo1a-eqp6rI2kF-afACZXlK3f3ymptZjV6M/edit?usp=sharing).
+
+## Conventions
+### Resource naming
+
+Every resource provisioned by Terraform should be created using the following naming schema:
+
+```
+`{client_shortname}-{environment}-{resource}-{additional_info}`
+```
+
+Examples:
+- `opencraftx-production-appserver-1` - Appserver from the production environment of client `OpenCraftX`
+- `example-stage-alb` - Alb resource for client `Example`
+
+To ease the migration and import process, the resources were renaming implies on changing credentials
+or replacement, a name override variable is available.
+
+The variables that allow name overrides are detailed on each module, and should be named as following:
+```
+override_{resource_name}_name
+```
 
+### Module categories & repo folder structure
+
+The Terraform modules must be categorized in one of the following categories:
+
+- `Core`: used to provision a vanilla Open edX instance and it's **required** dependencies, such
+as S3 buckets, databases, load balancer, security groups, etc.
+- `Optional`: depend on resources provisioned by the core module and are used to automate parts of
+the provisioning that might change from a customer to another one, as well as deploy additional
+services and functionality to the platform. Examples of modules here are: DNS management, email
+provisioning, marketing websites, auto-scaling and analytics.
+
+These categories dictate the directory a given module will live in:
+```
+# Main Open edX services are provisioned here
+/modules/core/openedx
+
+# Other modules should live here
+/modules/optional/analytics
+/modules/optional/marketing_site
+/modules/optional/ses_email_setup
+```
+
+This folder structure serves two purposes:
+1. Allow developers to easily differentiate between optional and core services to an Open edX instance.
+2. Accomodate for differences in deployment types between clients without making modules unecessarily
+complex with toggles and settings.
+
+### State storage
+
+The Terraform state file should be stored on a S3 bucket in the client's AWS account.
```

Comment:
This is fine with me, though now that I've tried GitLab's terraform backend I think it's even better. The main advantages of the GitLab backend are that it supports locking out of the box and it doesn't require any configuration.

If the client's private Terraform repo is on GitLab I would definitely suggest checking out their backend and also running Terraform via GitLab CI instead of locally - you can see examples of both in my Tutorraform project.

Reply:
Neat! I've added gitlab and s3 as acceptable state storage backends.

The suggestion about the gitlab terraform runner will come up in a follow up ticket to write usage/migration/import documentation (https://docs.google.com/document/d/1Kg_3pugyoXp6hiPeWyEy7hq-r984mE6iUkaRPOGnJ5c/edit?disco=AAAAIOVCmYo).